### PR TITLE
[IoT] Deprecate `az iot hub route` and `az iot hub routing-endpoint`

### PR DESCRIPTION
**Pull request by vilit1**
**Related command**
`az iot hub route`
`az iot hub routing-endpoint`

**Description**
Deprecation of  `az iot hub route` and `az iot hub routing-endpoint`. Users will use the azure iot cli extension commands `az iot hub message-route` and `az iot hub message-endpoint` respectively.

**Testing Guide**
None

**History Notes**
[IoT] `az iot hub route`: Hide the deprecated command, please use `az iot hub message-route` instead of it.
[IoT] `az iot hub routing-endpoint`: Hide the deprecated command, please use `az iot hub message-endpoint` instead of it.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
IoT

**Review by digimaun — DISMISSED** [review]

**Review by c-ryan-k — DISMISSED** [review]

**Review by digimaun — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 1]
```diff
@@ -162,7 +164,8 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('renew-key', 'iot_hub_policy_key_renew', supports_no_wait=True)
 
     # iot hub routing endpoint commands
-    with self.command_group('iot hub routing-endpoint', client_factory=iot_hub_service_factory) as g:
+    with self.command_group('iot hub routing-endpoint', client_factory=iot_hub_service_factory,
+                            deprecate_info=self.deprecate(redirect=ENDPOINT_DEPRECATION_INFO)) as g:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:168]:
```suggestion
                            deprecate_info=self.deprecate(redirect=ENDPOINT_DEPRECATION_INFO, hide=True)) as g:
```

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 2]
```diff
@@ -179,7 +182,8 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('update', 'iot_message_enrichment_update')
 
     # iot hub route commands
-    with self.command_group('iot hub route', client_factory=iot_hub_service_factory) as g:
+    with self.command_group('iot hub route', client_factory=iot_hub_service_factory,
+                            deprecate_info=self.deprecate(redirect=ROUTE_DEPRECATION_INFO)) as g:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:186]: Same as above

File: src/azure-cli/azure/cli/command_modules/iot/commands.py (hunk 1 above)
Comment (vilit1) [reply src/azure-cli/azure/cli/command_modules/iot/commands.py:168]: I see no issues - will hide both

Comment (vilit1) [reply src/azure-cli/azure/cli/command_modules/iot/commands.py]: Added the hide to both and also a command that we had marked as deprecated for a while

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 3]
```diff
@@ -128,7 +130,8 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('create', 'iot_hub_create', supports_no_wait=True)
         g.custom_command('list', 'iot_hub_list')
         g.custom_command('show-connection-string', 'iot_hub_show_connection_string',
-                         deprecate_info=self.deprecate(redirect=CS_DEPRECATION_INFO))
+                         deprecate_info=self.deprecate(redirect=CS_DEPRECATION_INFO),
+                         hide=True)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:134]:
```suggestion
                         deprecate_info=self.deprecate(redirect=CS_DEPRECATION_INFO, hide=True),
                        )
```

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 4]
```diff
@@ -162,7 +165,9 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('renew-key', 'iot_hub_policy_key_renew', supports_no_wait=True)
 
     # iot hub routing endpoint commands
-    with self.command_group('iot hub routing-endpoint', client_factory=iot_hub_service_factory) as g:
+    with self.command_group('iot hub routing-endpoint', client_factory=iot_hub_service_factory,
+                            deprecate_info=self.deprecate(redirect=ENDPOINT_DEPRECATION_INFO),
+                            hide=True) as g:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:170]:
```suggestion
                            deprecate_info=self.deprecate(redirect=ENDPOINT_DEPRECATION_INFO, hide=True),
                            ) as g:
```

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 5]
```diff
@@ -179,7 +184,9 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('update', 'iot_message_enrichment_update')
 
     # iot hub route commands
-    with self.command_group('iot hub route', client_factory=iot_hub_service_factory) as g:
+    with self.command_group('iot hub route', client_factory=iot_hub_service_factory,
+                            deprecate_info=self.deprecate(redirect=ROUTE_DEPRECATION_INFO),
+                            hide=True) as g:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:189]:
```suggestion
                            deprecate_info=self.deprecate(redirect=ROUTE_DEPRECATION_INFO, hide=True),
                            ) as g:
```

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 6]
```diff
@@ -179,7 +182,8 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('update', 'iot_message_enrichment_update')
 
     # iot hub route commands
-    with self.command_group('iot hub route', client_factory=iot_hub_service_factory) as g:
+    with self.command_group('iot hub route', client_factory=iot_hub_service_factory,
+                            deprecate_info=self.deprecate(redirect=ROUTE_DEPRECATION_INFO), hide=True) as g:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:186]:
```suggestion
                            deprecate_info=self.deprecate(redirect=ROUTE_DEPRECATION_INFO, hide=True)) as g:
```

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 7]
```diff
@@ -128,7 +130,7 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('create', 'iot_hub_create', supports_no_wait=True)
         g.custom_command('list', 'iot_hub_list')
         g.custom_command('show-connection-string', 'iot_hub_show_connection_string',
-                         deprecate_info=self.deprecate(redirect=CS_DEPRECATION_INFO))
+                         deprecate_info=self.deprecate(redirect=CS_DEPRECATION_INFO), hide=True)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:133]:
```suggestion
                         deprecate_info=self.deprecate(redirect=CS_DEPRECATION_INFO, hide=True))
```

File: src/azure-cli/azure/cli/command_modules/iot/commands.py [hunk 8]
```diff
@@ -162,7 +164,8 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.custom_command('renew-key', 'iot_hub_policy_key_renew', supports_no_wait=True)
 
     # iot hub routing endpoint commands
-    with self.command_group('iot hub routing-endpoint', client_factory=iot_hub_service_factory) as g:
+    with self.command_group('iot hub routing-endpoint', client_factory=iot_hub_service_factory,
+                            deprecate_info=self.deprecate(redirect=ENDPOINT_DEPRECATION_INFO), hide=True) as g:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:168]:
```suggestion
                            deprecate_info=self.deprecate(redirect=ENDPOINT_DEPRECATION_INFO, hide=True)) as g:
```

File: src/azure-cli/azure/cli/command_modules/iot/commands.py (hunk 6 above)
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/commands.py:186]:
```suggestion
                            deprecate_info=self.deprecate(redirect=ROUTE_DEPRECATION_INFO, hide=True)) as g:
```

**Review by zhoxing-ms — COMMENTED** [review]
Please note that this is not a problem caused by line breaks, but rather a problem of misplaced parentheses

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**